src/font/system.rs

```diff
@@ -132,6 +133,12 @@ impl FontSystem {
     }
 
     pub fn get_font_matches(&mut self, attrs: Attrs<'_>) -> Arc<Vec<FontMatchKey>> {
+        // Empty the cache first if it reached the size limit
+        if self.font_matches_cache.len() >= Self::FONT_MATCHES_CACHE_SIZE_LIMIT {
+            log::trace!("empty font mache cache");
+            self.font_matches_cache = Default::default();
+        }
+
         self.font_matches_cache
             //TODO: do not create AttrsOwned unless entry does not already exist
             .entry(AttrsOwned::new(attrs))
```

Comment:
I just realized this will insert a new item if something unrelated to finding fonts changes (like the metadata or color)

Reply:
Indeed.

Do you think it's worth it to change the map key to only contain relevant `Attrs` info?

Reply:
I do think that is important. Can make a new struct specifically for this map's key

Reply:
Done.